modules/cloud-credential-operator.adoc

```diff
@@ -8,8 +8,290 @@
 [discrete]
 == Purpose
 
-The Cloud Credential Operator manages cloud provider credentials as Kubernetes
-custom resource definitions (CRDs).
+The Cloud Credential Operator (CCO) manages cloud provider credentials as
+Kubernetes custom resource definitions (CRDs). The CCO syncs on
+`credentialsRequest` custom resources (CRs) to allow {product-title} components
+to request cloud provider credentials with the specific permissions that are
+required for installation and other tasks.
+
+[discrete]
+=== Modes
+
+The CCO can be configured to operate in _mint_, _passthrough_, or _manual_ mode.
+These options provide transparency and flexibility in how the CCO uses cloud
+credentials to process `credentialsRequest` CRs in the cluster, and allow the
+CCO to be configured to suit an organization's security requirements.
```

Comment:
aaah ok that detail was lost on me. Thanks for clarifying!

Reply:
> failing even passthrough mode means the credentials are deemed insufficient for CCO to operate with.

Then what happens? Does it complain? If the user can't provide adequate creds, do they de facto have to manually create IAM?

Reply:
That depends on when the failure happens. If it is during installation, the installer will call into some CCO code to make sure that mint or passthrough can work (this is only on AWS), and it will error very early in the install telling you which permissions were missing.

If after installation, the credentials are changed and CCO marks the creds as "insufficient", then CCO will put conditions on any new CredentialsRequest to indicate that it cannot process them because of the insufficient creds.

The user can fix the credentials, and CCO should eventually re-try to process the CredentialsRequest. Or the user can move CCO to manual/disabled mode and then handle the CredentialsRequest themselves.